subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy

```diff
@@ -995,7 +995,6 @@ task generate(type: TransformerTask) {
 """
 
         when:
-        executer.expectDocumentedDeprecationWarning(":transform1 consumes the output of :src2, but does not declare a dependency. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. Execution optimizations are disabled due to the failed validation. See https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.")
```

Comment:
Not about this line change here, but the message: I think this should say "task `:src2`" instead of just `:src2`, i.e. we should use the display name instead of the path of the task.

Reply:
I think the error message should explain a bit more what is happening. Something around those lines:

> The task :transform1 uses the output of another task, :src2, without declaring an explicit dependency (using `dependsOn`) or an implicit dependency (declaring `:src2` as an input). This will cause correctness issues, depending on what order the tasks are executed. This behavior has been deprecated and is scheduled to be removed in Gradle 7.0. Execution optimizations are disabled due to the failed validation. See https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.

Reply:
I changed the error message to something similar to what @melix suggested.

@lptr I can't access the `toString()` method of `task` here directly (task is not describable) and I tried changing the `toString()` method of local task node, resulting in quite a few test failures. Some of the messages we currently print are better off using the identity path of the task, so if we want to make the change to use Task's `toString()` in `LocalTaskNode`, I would prefer doing that in a separate PR.

Reply:
Fair enough. Let's do this when we revamp the error message.